perfkitbenchmarker/providers/gcp/gce_virtual_machine.py

```diff
@@ -48,26 +49,95 @@
 WINDOWS_IMAGE = 'windows-2012-r2'
 
 
+class MemoryDecoder(option_decoders.StringDecoder):
+  """Verifies and decodes a config option value specifying a memory size.
+
+  Attributes:
+    component: string. Description of the component to which the option applies.
+    option: string. Name of the config option.
+    required: boolean. True if the config option is required. False if not.
+  """
+
+  _CONFIG_MEMORY_PATTERN = re.compile(r'([0-9.]+)([GM]B)')
+
+  def Decode(self, value):
+    """Decodes memory size in MB from a string.
+
+    The value specified in the config must be a string representation of the
+    memory size expressed in MB or GB. It must be an integer number of MB
+    Examples: "1280MB", "7.5GB".
+
+    Args:
+      value: The value specified in the config.
+
+    Returns:
+      int. Memory size in MB.
+
+    Raises:
+      errors.Config.InvalidValue upon invalid input value.
+    """
+    string = super(MemoryDecoder, self).Decode(value)
+    match = self._CONFIG_MEMORY_PATTERN.match(string)
+    if not match:
+      raise errors.Config.InvalidValue(
+          'Invalid {0} "{1}" value: "{2}". Examples of valid values: '
+          '"1280MB", "7.5GB".'.format(self.component, self.option, string))
+    try:
+      memory_value = float(match.group(1))
+    except ValueError:
+      raise errors.Config.InvalidValue(
+          'Invalid {0} "{1}" value: "{2}". "{3}" is not a valid '
+          'float.'.format(self.component, self.option, string, match.group(1)))
+    memory_units = match.group(2)
+    if memory_units == 'GB':
+      memory_value *= 1024
+    memory_mb_int = int(memory_value)
+    if memory_value != memory_mb_int:
+      raise errors.Config.InvalidValue(
+          'Invalid {0} "{1}" value: "{2}". The specified size must be an '
+          'integer number of MB.'.format(self.component, self.option, string))
+    return memory_mb_int
+
+
 class GceVmSpec(virtual_machine.BaseVmSpec):
-  """Object containing the information needed to create a GceVirtualMachine."""
+  """Object containing the information needed to create a GceVirtualMachine.
+
+  Attributes:
+    cpus: None or int. Number of vCPUs for custom VMs.
+    install_packages: boolean. If False, no packages will be installed. This is
+        useful if benchmark dependencies have already been installed.
+    image: string or None. The disk image to boot from.
+    machine_type: string or None. The instance type name for non-custom VMs.
+        Example: 'n1-standard-8'.
+    memory: None or string. For custom VMs, a string representation of the size
+        of memory, expressed in MB or GB. Must be an integer number of MB (e.g.
+        "1280MB", "7.5GB").
+    num_local_ssds: int. The number of local SSDs to attach to the instance.
+    preemptible: boolean. True if the VM should be preemptible, False otherwise.
+    project: string or None. The project to create the VM in.
+    zone: string or None. The region / zone in which to launch the VM.
```

Comment:
Personally, I'd prefer to just document the attributes related to the GceVmSpec specifically here and document the other ones in the BaseVmSpec. It shouldn't be that hard to keep them in sync though since I don't expect the BaseVmSpec ones to change that much.